### ISSUE TITLE
initial_root_password attribute cannot handle blank password

### ISSUE THREAD

**Issue by webbower** [chef-cookbooks/mysql#340]
Given the following code in a recipe:

``` ruby
mysql_service 'default' do
  initial_root_password '' # Blank password. For local dev purposes
  action [:create, :start]
end
```

When it reaches the following line:

https://github.com/chef-cookbooks/mysql/blob/a1466f6e20e254bf4cdfa26c98216d1a357d6b96/libraries/helpers.rb#L152

`Shellwords.escape('')` (and `Shellwords.escape("")`) returns the string `"''"`. That's a string of length 2 comprised of 2 single quotes. And in the context of the above line, that messes with the string boundaries for the password. Seems to be a quirk of that Ruby function.

A couple suggestions:
* Change the string boundary to double quotes to protect against this (if double quotes are allowed in SQL in that context).
* Check if the `new_resource.initial_root_password` is empty string and handle specially in that case.
* Explicitly disallow blank root passwords and throw an error (I dislike this idea since blank root passwords are handy for local devving. If you keep them allowed, maybe output a warning so people are aware of the potential vulnerability.).

**Comment by webbower** [comment]
Ok. Further delving. I thought that behavior of `Shellwords.escape` seemed odd, but it's apparently explicitly handled that way in 2.0 and 2.1 https://github.com/ruby/ruby/blob/v2_0_0_645/lib/shellwords.rb#L127

**Comment by someara** [comment]
Fixed in 6.0.29
